client/src/App.js

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { BrowserRouter as Router } from "react-router-dom";
+import { Navbar } from "./components/Navbar";
+import { useRoutes } from "./routes";
+import { useAuth } from "./hooks/auth.hook";
+import { AuthContext } from "./context/AuthContext";
+
+import Modal from "react-modal";
+import "materialize-css";
+
+Modal.setAppElement("#root");
+
+function App() {
+  const { token, login, logout, userId } = useAuth();
+  const isAuthenticated = !!token;
+  const routes = useRoutes(isAuthenticated);
+  return (
+    <AuthContext.Provider
+      value={{
+        token,
+        login,
+        logout,
+        userId,
+        isAuthenticated,
+      }}
+    >
+      <Router>
+        <Navbar />
+
+        <div className="container">{routes}</div>
+      </Router>
+    </AuthContext.Provider>
+  );
+}
```

Comment:
I would simplify this piece of code like on my image
![AuthProvider](https://user-images.githubusercontent.com/51495533/125253843-82dea080-e302-11eb-9e38-f8913d64edff.png)